src/components/richText/imageNode/ImageNode.tsx

```diff
@@ -0,0 +1,200 @@
+import {
+  $applyNodeReplacement,
+  DOMConversionMap,
+  DOMConversionOutput,
+  DOMExportOutput,
+  EditorConfig,
+  DecoratorNode,
+  LexicalNode,
+  NodeKey,
+  SerializedLexicalNode,
+  Spread,
+} from "lexical";
+
+import * as React from "react";
+
+import LexicalImage from "./LexicalImage";
+
+/**
+ * The raw Object obtained when first parsing a serialized {@link ImageNode}.
+ */
+export type SerializedImageNode = Spread<
+  {
+    altText: string;
+    height?: number;
+    maxWidth: number;
+    src: string;
+    width?: number;
+    type: "image";
+    version: 1;
+  },
+  SerializedLexicalNode
+>;
+
+/**
+ * Defines a Lexical Dev {@link DecoratorNode} for images.
```

Comment:
Would it be possible to have more documentation here? Normally I prefer to not describe "how" something works via documentation and just have people read code, but this class has a lot of lexical specific methods that seem difficult to get a grasp on.